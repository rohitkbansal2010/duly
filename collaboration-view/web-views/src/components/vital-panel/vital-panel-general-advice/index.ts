export { VitalPanelGeneralAdvice } from './vital-panel-general-advice';
export type { VitalPanelGeneralAdvicePropsType } from './vital-panel-general-advice';
