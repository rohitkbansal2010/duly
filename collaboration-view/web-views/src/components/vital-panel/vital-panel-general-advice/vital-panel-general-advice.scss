@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.vitalPanelGeneralAdviceContainer {
  padding: 1.25rem 1.25rem 1.125rem;
  border-left: 0.06rem solid $iron;

  &Title {
    margin: 0;
    margin-bottom: 0.25rem;
    padding: 0;
    color: $medium-gray;
    font-size: var(--font-size-16px-4px);
    line-height: 1.625rem;
    @include nexa-heavy-highlight;
  }

  &Text {
    margin: 0;
    padding: 0;
    color: $abbey;
    font-size: var(--font-size-14px-4px);
    line-height: var(--line-height-16px-4px);
    @include nexa-regular-highlight;
  }
}
