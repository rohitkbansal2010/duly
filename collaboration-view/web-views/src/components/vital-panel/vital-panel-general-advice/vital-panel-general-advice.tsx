import { CSSProperties } from 'react';

import styles from './vital-panel-general-advice.scss';

export type VitalPanelGeneralAdvicePropsType = {
  title: string;
  text: string;
  style: CSSProperties;
};

export const VitalPanelGeneralAdvice = ({
  title,
  text,
  style,
}: VitalPanelGeneralAdvicePropsType) =>
  (
    <div className={styles.vitalPanelGeneralAdviceContainer} style={style}>
      <h2
        className={styles.vitalPanelGeneralAdviceContainerTitle}
        data-test="general-advice-title"
      >
        {title}
      </h2>
      <p
        className={styles.vitalPanelGeneralAdviceContainerText}
        data-test="general-advice-text"
      >
        {text}
      </p>
    </div>
  );
