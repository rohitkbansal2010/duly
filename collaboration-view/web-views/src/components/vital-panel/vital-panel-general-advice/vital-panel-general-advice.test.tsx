import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { VitalPanelGeneralAdvice, VitalPanelGeneralAdvicePropsType } from './vital-panel-general-advice';

const mockProps: VitalPanelGeneralAdvicePropsType = {
  title: 'General Advice',
  text: 'Blood pressure numbers of less than 120/80 mm Hg are considered within the normal range. If your results fall into this category, stick with heart-healthy habits.',
  style: { width: '56%' },
};

describe('VitalPanelGeneralAdvice', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<VitalPanelGeneralAdvice {...mockProps} />);
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render title', () => {
    const titleElement = component.find('[data-test="general-advice-title"]');

    expect(titleElement).toHaveLength(1);
    expect(titleElement.text().includes(mockProps.title)).toBeTruthy;
  });

  it('should render text', () => {
    const textElement = component.find('[data-test="general-advice-text"]');

    expect(textElement).toHaveLength(1);
    expect(textElement.text().includes(mockProps.text)).toBeTruthy;
  });
});
