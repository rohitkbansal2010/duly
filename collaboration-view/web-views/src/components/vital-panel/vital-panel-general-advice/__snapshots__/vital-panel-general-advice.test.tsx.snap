// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VitalPanelGeneralAdvice should render correctly 1`] = `
<div
  className="vitalPanelGeneralAdviceContainer"
  style={
    Object {
      "width": "56%",
    }
  }
>
  <h2
    className="vitalPanelGeneralAdviceContainerTitle"
    data-test="general-advice-title"
  >
    General Advice
  </h2>
  <p
    className="vitalPanelGeneralAdviceContainerText"
    data-test="general-advice-text"
  >
    Blood pressure numbers of less than 120/80 mm Hg are considered within the normal range. If your results fall into this category, stick with heart-healthy habits.
  </p>
</div>
`;
